### PR TITLE
🐛  #1111 Handle Empty ClinicalRegistrationData Records

### PR DESCRIPTION
**Pull request by demariadaniel**
## Link to Issue

[#1111 ](https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/roadmap/1111)

## Description

Handles empty values for `ClinicalRegistrationData.records` with default []
Refactor variable definitions for readability

### Type of Change

- [x] Bug
- [x] Refactor
- [ ] New Feature
- [ ] Release Candidate

### Checklist before requesting review:

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/schemas/utils.ts [hunk 1]
```diff
@@ -33,265 +33,268 @@ const ARRAY_DELIMITER_CHAR = '|';
...
-    batchErrors?: { message: string; batchNames: string[]; code: string }[];
-  },
+	programShortName: string,
+	data: {
+		registration: DeepReadonly<ActiveRegistration> | undefined;
+		errors?: RegistrationErrorData[];
+		batchErrors?: { message: string; batchNames: string[]; code: string }[];
+	},
 ) => {
-  const registration: Partial<typeof data.registration> = get(data, 'registration', {});
```
Comment (demariadaniel) [inline src/schemas/utils.ts:91]: Actual changes (non-prettier) are this block

**Comment by ciaranschutte** [comment]
@demariadaniel please let me know which line change is on. prettier has formatted the entire file so I'm not sure where the diff is.

File: src/schemas/utils.ts [hunk 2]
```diff
@@ -19,10 +19,10 @@
 
 import get from 'lodash/get';
 import {
-  ActiveRegistration,
```
Comment (demariadaniel) [inline src/schemas/utils.ts:22]: Large # of Prettier changes

File: src/schemas/utils.ts (hunk 1 above)
Comment (demariadaniel) [reply src/schemas/utils.ts:91]: cc: @ciaranschutte

File: src/schemas/utils.ts [hunk 3]
```diff
@@ -33,265 +33,268 @@ const ARRAY_DELIMITER_CHAR = '|';
...
-    newSpecimens: () =>
-      convertRegistrationStatsToGql(get(registration, 'stats.newSpecimenIds', [])),
-    newSamples: () => convertRegistrationStatsToGql(get(registration, 'stats.newSampleIds', [])),
-    alreadyRegistered: () =>
-      convertRegistrationStatsToGql(get(registration, 'stats.alreadyRegistered', [])),
-  };
+	const registration: Partial<typeof data.registration> = get(data, 'registration', {});
+	const schemaAndValidationErrors: typeof data.errors = get(data, 'errors', []);
+	const fileErrors: typeof data.batchErrors = get(data, 'batchErrors', []);
+	const records = get(registration, 'records', []);
```
Comment (demariadaniel) [inline src/schemas/utils.ts:94]: Adding this empty array to `records` is the only change needed to fix the issue

**Comment by demariadaniel** [comment]
> @demariadaniel please let me know which line change is on. prettier has formatted the entire file so I'm not sure where the diff is.

see: https://github.com/icgc-argo/argo-clinical/pull/1137#discussion_r1489988825

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**